Something/WebApplication/wwwroot/js/apps/organization/components/CreateOrganization.tsx

```diff
@@ -0,0 +1,100 @@
+    public static displayName = "CreateOrganization";
+    constructor(props: CreateOrganization.Props) {
+        super(props);
+        this.state = {
+            hasError: false,
+            organizationName: "",
+        };
+        this.onChangeOrgnaizationName = this.onChangeOrgnaizationName.bind(this);
+        this.createOrganization = this.createOrganization.bind(this);
+    }
+
+    public render(): JSX.Element {
+        if (!this.state) { return null; }
+        return (
+            <div className="ant-row">
+                {this.renderForm()}
+                <hr className="ant-col-md-24 CreateOrganization__hr"/>
+                {this.renderError()}
+            </div>
+        );
+    }
+
+    private renderForm(): JSX.Element {
+        return <Form
+            className="ant-col-md-24 CreateOrganization"
+        >
+            <p>
+                <b>
+                    Get most out of your organization
+                </b>
+            </p>
+            <p>Organization helps you to group users and projects. You can create mulitple organization, one user can be linked with mulitple organizations.</p>
+            <b>Organization name</b>
+            <FormItem>
+                <Input
+                    className="ant-input-lg CreateOrganization__input"
+                    placeholder="ACME INC"
+                    name="Email"
+                    onChange={this.onChangeOrgnaizationName}
+                    autoFocus
+                />
+            </FormItem>
+            <FormItem>
+                <Button
+                    type="primary"
+                    htmlType="submit"
+                    className={this.getButtonCssClasses()}
+                    onClick={this.createOrganization}
+                >
+                    Create Organization
+                        </Button>
+            </FormItem>
+        </Form>;
+    }
+
+    private renderError(): JSX.Element {
+        return this.state.hasError ?
+            <Alert
+                className="ant-col-md-24"
+                message="Oops! Something went wrong."
+                description="There was an error creating your orgranization. Please try again in sometime."
+                type="error"
+            /> : null;
+    }
+
+    private getButtonCssClasses(): string {
+        return classNames("login-form-button", "ant-btn--rounded",
+            { "ant-btn--disabled": !(this.state.organizationName && this.state.organizationName.length >= 3) },
+        );
+    }
+
+    private onChangeOrgnaizationName(e: any) {
+        this.setState({ organizationName: e.target.value });
+    }
+
+    private createOrganization() {
+        const that = this;
+        OrganizationService.post(this.state.organizationName)
+            .then((response) => response.text())
+            .then((organizationId) => that.onOrganizationCreated(organizationId))
+            .catch((response) => that.setState({hasError: true}));
```

Comment:
Function setState

Reply:
Functional setState to always use or in specific cases?

Reply:
Do it always, follow one style